### DISCUSSION TITLE
There's no built-in way to rename an argument. Plus ﻿`metavar` param probably should change only the type representation of an argument.

### DISCUSSION THREAD

**Discussion by vladhaidukkk** [fastapi/typer#845] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typer import Typer

app = Typer()


@app.command()
def complete_task(id_: int):  # I want it to be shown as id & ID in the help
    pass
```

### Description

I need a way to rename the argument from **id_** to **id** in the help message. Additionally, it should be displayed in uppercase in the usage section. I've found a recommendation to use `metavar` for this, but it changes the type representation of the argument and how it's displayed in the usage section, which is not what I need.

I'd like to have a `name` parameter for this in the `Argument`.

### Operating System

macOS

### Typer Version

0.12.3

### Python Version

3.12

**Comment by devinnasar** [comment]
I'm also looking to understand how to do this. I don't like that my arguments use underscores and my options use hyphens.

**Comment by Tmpod** [comment]
I ran into this and went looking how parameter names are passed to click.  

Ended up cooking this temporary (and very dirty) patch :sweat_smile: 

```python
def patch_param_names():
    """Call this before calling your Typer app."""
    import typer

    def get_params_from_function_patched(func):
        params = typer.utils.get_params_from_function(func)
        new_params = {}
        for name, meta in list(params.items()):
            if name.startswith("_") or name.endswith("_"):
                stripped = name.strip("_")
                new_params[stripped] = typer.models.ParamMeta(
                    name=stripped, default=meta.default, annotation=meta.annotation
                )
            else:
                new_params[name] = meta

        return new_params

    typer.main.get_params_from_function = get_params_from_function_patched
```

_Edit: my previous version would cause argument reordering_
